app/src/components/code/JSONEditor.tsx

```diff
@@ -1,24 +1,56 @@
-import React from "react";
-import { json, jsonParseLinter } from "@codemirror/lang-json";
+import React, { useMemo } from "react";
+import { json, jsonLanguage, jsonParseLinter } from "@codemirror/lang-json";
 import { linter } from "@codemirror/lint";
-import { EditorView } from "@codemirror/view";
+import { EditorView, hoverTooltip } from "@codemirror/view";
+import { githubLight } from "@uiw/codemirror-theme-github";
 import { nord } from "@uiw/codemirror-theme-nord";
 import CodeMirror, { ReactCodeMirrorProps } from "@uiw/react-codemirror";
+import {
+  handleRefresh,
+  jsonCompletion,
+  jsonSchemaHover,
+  jsonSchemaLinter,
+  stateExtensions,
+} from "codemirror-json-schema";
+import { JSONSchema7 } from "json-schema";
```

Comment:
what's the 7 here?

Reply:
the version of the json schema spec i believe